pkg/manager/impl/validators/partition_validator.go

```diff
@@ -0,0 +1,83 @@
+package validators
+
+import (
+	"fmt"
+
+	"github.com/lyft/datacatalog/pkg/errors"
+	datacatalog "github.com/lyft/datacatalog/protos/gen"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	partitionKeyName   = "partitionKey"
+	partitionValueName = "partitionValue"
+)
+
+func ValidatePartitions(datasetPartitionKeys []string, artifactPartitions []*datacatalog.Partition) error {
+	if len(datasetPartitionKeys) != len(artifactPartitions) {
+		return errors.NewDataCatalogErrorf(codes.InvalidArgument, "Partition key mismatch, dataset keys: %+v, artifact Partitions: %+v", datasetPartitionKeys, artifactPartitions)
+	}
+
+	// Not all datasets need to be partitioned
```

Comment:
Why not?

Reply:
These are like labels in Catalog V1, e.g. `region`, `ds`.  If no such partitions are needed by the user why would we want to force them to use one?

Reply:
Gotcha. That makes sense.